src/refiners/foundationals/latent_diffusion/stable_diffusion_1/unet.py

```diff
@@ -188,7 +188,7 @@ def __init__(
                 fl.UseContext(context="unet", key="residuals").compose(lambda x: x[-1]),
                 MiddleBlock(device=device, dtype=dtype),
             ),
-            UpBlocks(),
+            UpBlocks(device=device, dtype=dtype),
```

Comment:
Nice catch. Could you please add a test (for coverage) under `tests/foundationals/latent_diffusion/test_unet.py`?

(I guess in practical cases, we benefited from `self.unet = unet.to(device=self.device, dtype=self.dtype)` in `LatentDiffusionModel` CTOR)

Reply:
yes indeed the `LatentDiffusionModel`'s init fixed the problem, this problem arises when testing/using the bare unet. I'll add a simple "forward test".